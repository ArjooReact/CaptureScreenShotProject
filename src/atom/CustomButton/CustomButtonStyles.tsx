import {StyleSheet} from 'react-native';
import Colors from '../../utility/Colors';
import Dimension from '../../utility/Dimension';
const styles = StyleSheet.create({
  mainContainer: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    width: Dimension.CustomButtonWidth,
    height: Dimension.CustomButtonHeight,
    backgroundColor: Colors.SKY_BLUE,
    borderRadius: Dimension.BorderRadius,
    marginTop: Dimension.CustomButtonMarginTop,
  },
  fontStyle: {
    fontSize: Dimension.FontSize,
    color: Colors.WHITE,
    fontWeight: 'bold',
  },
});

export default styles;
