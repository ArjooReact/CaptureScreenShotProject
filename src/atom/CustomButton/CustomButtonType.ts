export interface CustomButtonType{
    title:string
    onClickHandler:()=> void
}