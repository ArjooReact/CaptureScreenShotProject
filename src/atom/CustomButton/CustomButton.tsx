import React from 'react'
import { TouchableOpacity,SafeAreaView,Text, StyleSheet } from 'react-native'
import { CustomButtonType } from './CustomButtonType'
import styles from './CustomButtonStyles'

const CustomButton:React.FC<CustomButtonType>=({title,onClickHandler})=>{
return(<TouchableOpacity
style={styles.mainContainer}
onPress={onClickHandler}
>
    <Text style={styles.fontStyle}>{title}</Text>
</TouchableOpacity>)
}
export default CustomButton