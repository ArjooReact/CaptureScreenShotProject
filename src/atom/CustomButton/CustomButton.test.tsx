import renderer from 'react-test-renderer';
import {test, jest, describe} from '@jest/globals';
import {expect, it, beforeEach} from '@jest/globals';
import CustomButton from './CustomButton';

describe('all mock', () => {
  let mockButtonClick = jest.fn();

  test('testcase for matching Snapshot==>', () => {
    const AppRef = renderer.create(
      <CustomButton title="PostCall" onClickHandler={mockButtonClick} />,
    );
    expect(AppRef.toJSON()).toMatchSnapshot();
  });

  test('testcase for checking dom element for CustomButton==>', () => {
    const AppRef = renderer.create(
      <CustomButton title="PostCall" onClickHandler={mockButtonClick} />,
    );
    console.log('arzoo test======', AppRef.toJSON());
    let props: any = AppRef.toJSON();

    expect(AppRef.toJSON()).toBeTruthy();
  });

  test('testcase for checking child element for CustomButton==>', () => {
    const styles = {};
    const AppRef = renderer.create(
      <CustomButton title="PostCall" onClickHandler={mockButtonClick} />,
    );
    let props: any = AppRef.toJSON();
    expect(props?.children[0].children[0]).toBe('PostCall');
  });

  test('testcase for checking child element for CustomButton==>', () => {
    //const styles={}
    const AppRef: any = renderer.create(
      <CustomButton title="PostCall" onClickHandler={mockButtonClick} />,
    );
    const styles = AppRef.toJSON()?.props.style;
    console.log('Propsdd.ff....', styles);
    const {width, height, backgroundColor, borderRadius} = styles;
    expect(width).toBe(160);
    expect(height).toBe(60);
    expect(backgroundColor).toBe('#20a7db');
    expect(borderRadius).toBe(8);
  });
});
