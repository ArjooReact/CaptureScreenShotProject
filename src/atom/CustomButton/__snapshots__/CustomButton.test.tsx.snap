// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all mock testcase for matching Snapshot==> 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#20a7db",
      "borderRadius": 8,
      "display": "flex",
      "height": 60,
      "justifyContent": "center",
      "marginTop": 20,
      "opacity": 1,
      "width": 160,
    }
  }
>
  <Text
    style={
      {
        "color": "#ffffff",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    PostCall
  </Text>
</View>
`;
